tests/lib/external/snapd-testing-tools/tests/remote.exec/task.yaml

```diff
@@ -0,0 +1,39 @@
+summary: smoke test for the remote.exec tool
+
+# Amazon linux is skipped because no sshpass available
+systems: [-amazon-linux-*]
+
+prepare: |
+    tests.pkgs install sshpass
+
+restore: |
+    tests.pkgs remove sshpass
+    rm -rf remote.setup.cfg my-key my-key.pub  /home/tools-user-1/*  /home/tools-user-2/*
+
+execute: |
+    remote.exec --help | MATCH 'usage: remote.exec \[--user <USER>\] \[--pass <PASS>\] <CMD>'
+    remote.exec -h | MATCH 'usage: remote.exec \[--user <USER>\] \[--pass <PASS>\] <CMD>'
+
+    remote.setup config --host localhost --port 22 --user tools-user-1 --pass tools-user-1
+
+    remote.exec "touch testfile"
+    test -f /home/tools-user-1/testfile
+
+    remote.exec --user tools-user-2 --pass tools-user-2 "touch testfile"
+    test -f /home/tools-user-2/testfile
```

Comment:
We touch testfile for tools-user-1 but check tools-user-2, is this ok?

Reply:
I think it's ok, it's connecting through ssh to itself as test-user-2

Reply:
ah yes, I misread it. Thanks.